lib/lambda_ethereum_consensus/state_transition/operations.ex

```diff
@@ -4,7 +4,127 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
   """
 
   alias LambdaEthereumConsensus.StateTransition.{Accessors, Misc, Mutators, Predicates}
-  alias SszTypes.{Attestation, BeaconState, ExecutionPayload, Validator, Withdrawal}
+  alias LambdaEthereumConsensus.Utils.BitVector
+
+  alias SszTypes.{
+    Attestation,
+    BeaconState,
+    ExecutionPayload,
+    SyncAggregate,
+    Validator,
+    Withdrawal
+  }
+
+  @spec process_sync_aggregate(BeaconState.t(), SyncAggregate.t()) ::
+          {:ok, BeaconState.t()} | {:error, String.t()}
+  def process_sync_aggregate(
```

Comment:
I think the function is too long. We should split this up into multiple smaller functions.